src/api.ts

```diff
@@ -17,6 +18,10 @@ export default class Api {
         'Content-Type': 'application/json;charset=utf-8',
       },
     });
+    if (!response.ok) {
+      const errorResponse = (await response.json()) as ErrorResponse;
+      throw new Error(errorResponse.status_message);
+    }
     return (await response.json()) as T;
```

Comment:
Hmmm to avoid duplicating this await of response.json(). I'm wondering if we can move it up.

Reply:
What happens now if an error is thrown when making a GET call?

Reply:
![image](https://user-images.githubusercontent.com/99838921/233307669-2e573dc0-6ea0-4d29-a919-f06f496e8b58.png)
It would throw an error which can be catched but somehow, I am personally not sure why, the error object will be empty.

Reply:
Yeah, when console logging an error it will always output an empty object. Tbh I rather let the consumer of the lib figure out what they want to do with error handling instead of explicitly returning the status_message.

Reply:
Here is a stackoverflow link for reference https://stackoverflow.com/questions/18391212/is-it-not-possible-to-stringify-an-error-using-json-stringify/26199752#26199752